bucket-registry.go

```diff
@@ -29,54 +29,117 @@ import (
 // `realpath` or `fcntl(F_GETPATH)`) and use the canonical path as the key.
 // Unfortunately Go doesn't seem to have an API for that.
 
-var bucketRegistry = map[string][]*Bucket{} // Maps URL to slice of Buckets at that URL
-var bucketRegistryMutex sync.Mutex          // Thread-safe access to bucketRegistry
+type bucketRegistry struct {
+	byURL       map[string][]*Bucket
```

Comment:
Does this need to be a `[]*Bucket`, or is it just used for ref counting and so could be something like `[]bool` or `[]struct{}`.

Reply:
I replaced this with integer refcounting.